src/covid_model_seiir_pipeline/forecasting/data.py

```diff
@@ -229,22 +214,18 @@ def save_specification(self, specification: ForecastSpecification):
     def save_raw_covariates(self, covariates: pd.DataFrame, scenario: str, draw_id: int):
         io.dump(covariates, self.forecast_root.raw_covariates(scenario=scenario, draw_id=draw_id))
 
-    def load_raw_covariates(self, scenario: str, draw_id: int):
-        covariates = io.load(self.forecast_root.raw_covariates(scenario=scenario, draw_id=draw_id))
-        covariates['date'] = pd.to_datetime(covariates['date'])
-        return covariates
-
     def load_beta_params(self, draw_id: int) -> Dict[str, float]:
         df = io.load(self.regression_root.parameters(draw_id=draw_id))
         return df.set_index('params')['values'].to_dict()
 
+    def save_ode_params(self, ode_params: pd.DataFrame, scenario: str, draw_id: int):
+        io.dump(ode_params, self.forecast_root.ode_params(scenario=scenario, draw_id=draw_id))
+
     def save_components(self, forecasts: pd.DataFrame, scenario: str, draw_id: int):
         io.dump(forecasts, self.forecast_root.component_draws(scenario=scenario, draw_id=draw_id))
 
-    def load_components(self, scenario: str, draw_id: int) -> pd.DataFrame:
-        components = io.load(self.forecast_root.component_draws(scenario=scenario, draw_id=draw_id))
-        components['date'] = pd.to_datetime(components['date'])
-        return components.set_index(['location_id', 'date'])
+    def load_components(self, scenario: str, draw_id: int):
+        return io.load(self.forecast_root.component_draws(scenario=scenario, draw_id=draw_id))
```

Comment:
Did you consistently push the date conversion into `io.load`? If not you're asking for trouble IFF a non-typed implementation like CSV ever gets used

Reply:
I haven't done that yet.  It is very definitely on my list.

Reply:
OK. My admittedly poor memory is that you either have to do a bad heuristic ala `if 'date' in df` or leave it here.

Or just delete the non-typed backends, which is probably reasonable.

Reply:
I just realized that this code changed cause I deleted it (it's unused in the actual pipeline), and then added it back in because a test failed.  I'll revert to the original implementation.

Reply:
I think csv has to stick around for the final outputs.  I've been thinking about the long term switch.  I'm going to make a little perfomance logger in a PR soon and then do some experiments.  Feather looks like a nice alternative from the benchmarks I've seen.